pkg/fleetallocation/controller.go

```diff
@@ -288,3 +292,25 @@ func (c *Controller) allocate(f *stablev1alpha1.Fleet) (*stablev1alpha1.GameServ
 
 	return gs, nil
 }
+
+// patch the labels and annotations of an allocated GameServer with metadata from a FleetAllocation
+func (c *Controller) patchMetadata(gs *stablev1alpha1.GameServer, fam *stablev1alpha1.FleetAllocationMeta) {
```

Comment:
@markmandel do you think we should move this to `func(gs *stablev1alpha1.GameServer) MergeFleetMetadata(fam *stablev1alpha1.FleetAllocationMeta)` and use it 
```
if fam != nil {
		gsCopy.MergeFleetMetadata(fam)
	}
```

Reply:
Hmnnn. Interesting question.

I guess I'm more inclined to leave it - since it's functionality that is only used in this particular place. If it ends up being logic that needs to be shared, we could move it then.

Do you think it likely that we would need to have this logic be run in multiple places?

Reply:
no but it could be a function though since `c` is not used

Reply:
We were talking of the game server being able to update its own metadata, right?
Could this function be reused for that?

Reply:
Yes, that's what I had in mind but I guess it's safer to wait until we have a definite need.